### DISCUSSION TITLE
Crack size and domain size

### DISCUSSION THREAD

**Discussion by Sofia-Y** [neperfepx/neper#176] [Q&A] [answered]
Hello @rquey ,

The following command works fine (I suppose it uses 'Voronoi' tessellation as default):
```
$ neper -T -n 10 -id 1 -dim 2 -ori uniform -oridescriptor euler-bunge -transform "cut(ecylinder(0,0,0,0,0,1,1,0,0,0,1,0,0.5,0.05))" 
$ neper -M n10-id1.tess -elttype tri -format 'inp'
```
However, when I add -morpho 'gg' and -domain (see example below) I do not get the crack. I only get the square.
```
$ neper -T -n 10 -id 1 -dim 2 -ori uniform -oridescriptor euler-bunge -transform "cut(ecylinder(0,0,0,0,0,1,1,0,0,0,1,0,0.5,0.05))" -domain 'square(330,330)' -morpho 'gg'
$ neper -M n10-id1.tess -elttype tri -format 'inp'
```

Can I get my crack deformation using the graingrowth tessellation and specifying my domain or does the -transform command works only with Voronoi and the default domain? 

I appreciate your help and thank you for this platform.

**Comment by rquey** [comment, marked as answer]
It does not come from the use of `graingrowth`. It would give the same results with a standard Voronoi tessellation.

The problem comes from the fact that your domain is much bigger in the 2nd case. The ecylinder is therefore very small with respect to the grain size, which is not well supported by the current implementation. Do you really want a crack that small in such a large domain?

**Reply by Sofia-Y** [reply]
So what is the default domain here, with the graingrowth? I tried -domain 'square(10,10)' and it did not work either. My aim is to create a noticeable crack, not really a small one. Would you suggest I keep changing the -domain or the -transform cut?

Thank you.

**Reply by rquey** [reply]
The default domain is always a unit domain (1x1 or 1x1x1). You need to work out the domain and ecylinder sizes (and the number of grains). If the ecylinder is orders of magnitude smaller, you just won't see it. 

```
$ neper -T -n 1000 -id 1 -dim 2 -oridescriptor euler-bunge -transform "cut(ecylinder(0,0,0,0,0,1,1,0,0,0,1,0,5,0.5))" -domain 'square(10,10)' -o s10
$ neper -V s10.tess -datacellcol id -print s10
```
![image](https://user-images.githubusercontent.com/2659376/114605773-b88bd300-9c9a-11eb-8c0f-fc23d7e76571.png)

**Reply by Sofia-Y** [reply]
Thank you!

**Comment by Sofia-Y** [comment]
Hello, @rquey !

Just a side note on this example; below are 2 same exact examples one with Voronoi grains and one with 'gg'. They do not give the same results ( I am just double-checking with you as you have mentioned that the issue was not from using 'gg'. I am a little confused here, any clarification would be appreciated. Thank you.

•	neper -T -n 10 -id 1 -dim 2 -oridescriptor euler-bunge -transform "cut(ecylinder(0,50,0,0,0,1,1,0,0,0,1,0,50,5))" -domain 'square(100,100)' 
•	neper -M n10-id1.tess -elttype tri -format 'inp' 

![Picture1](https://user-images.githubusercontent.com/56741948/114785358-3656e980-9d42-11eb-9f84-92899aabf784.png)
 
•	neper -T -n 10 -id 1 -dim 2 -oridescriptor euler-bunge -transform "cut(ecylinder(0,50,0,0,0,1,1,0,0,0,1,0,50,5))" -domain 'square(100,100)' -morpho 'gg'
•	neper -M n10-id1.tess -elttype tri -format 'inp' 

![Picture2](https://user-images.githubusercontent.com/56741948/114785467-5e464d00-9d42-11eb-8b6f-177e49897c1d.png)

**Reply by rquey** [reply]
This is because the ecylinder characteristic size is close to or smaller than the grain size. This is a limitation of the current algorithm and it is _maybe_ more sensitive with gg than with a standard Voronoi tessellation. For a larger number of grains, you will see that the problem disappears.

**Reply by Sofia-Y** [reply]
Thanks a lot for your time!
